modules/core/src/main/java/org/apache/ignite/internal/util/ByteUtils.java

```diff
@@ -64,6 +65,17 @@ public static long bytesToLong(byte[] bytes) {
         return bytesToLong(bytes, 0);
     }
 
+
+    /**
+     * Constructs {@code long} from byte array created with {@link #longToComparableBytes(long)} with offset equal to 0.
+     *
+     * @param bytes Array of bytes.
+     * @return Long value.
+     */
+    public static long comparableBytesToLong(byte[] bytes) {
```

Comment:
What's the penalty of such serialisation/deserialisation in comparison to `longToBytes` that forces you to have both? I mean that there's a chance that somebody will miss the fact the there's an extra comparable version of the method and will use common one where he actually need comparison.

Reply:
No penalty at all. If developers won't try deserializing data by hand and will always use these methods, then there would be no difference at all

Reply:
So, my question is actually following. Should we include xoring into common longToBytes, intToBytes, etc in order to preserve comparison in lex-order and remove comparable<> counterparts that you've added?

Reply:
What if someone wants to get a BE representation specifically?

Reply:
We already have instances where these methods are used exclusively as a conversion to BE. What do we do with them?

Reply:
Discussed in private. Up to you.